packages/react-reconciler/src/ReactFiberNewContext.js

```diff
@@ -34,21 +36,38 @@ export default function(stack: Stack) {
   function pushProvider(providerFiber: Fiber): void {
     const context: ReactContext<any> = providerFiber.type._context;
 
-    push(changedBitsCursor, context._changedBits, providerFiber);
-    push(valueCursor, context._currentValue, providerFiber);
-    push(providerCursor, providerFiber, providerFiber);
-
-    context._currentValue = providerFiber.pendingProps.value;
-    context._changedBits = providerFiber.stateNode;
-
-    if (__DEV__) {
-      warning(
-        context._currentRenderer === null ||
-          context._currentRenderer === rendererSigil,
-        'Detected multiple renderers concurrently rendering the ' +
-          'same context provider. This is currently unsupported.',
-      );
-      context._currentRenderer = rendererSigil;
+    if (isPrimaryRenderer) {
+      push(changedBitsCursor, context._changedBits, providerFiber);
+      push(valueCursor, context._currentValue, providerFiber);
+      push(providerCursor, providerFiber, providerFiber);
+
+      context._currentValue = providerFiber.pendingProps.value;
+      context._changedBits = providerFiber.stateNode;
+      if (__DEV__) {
+        warning(
+          context._currentRenderer === null ||
+            context._currentRenderer === rendererSigil,
+          'Detected multiple renderers concurrently rendering the ' +
```

Comment:
I'm bumping into this with test renderer when the context object is reused, but `resetModules()` is called between tests. Seems confusing.